lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1129,8 +1152,10 @@ def _can_service_start(self) -> bool:
             not self.peers_data.get(Scope.APP, "security_index_initialised", False)
             or not self.alt_hosts
         ):
-            return (
-                self.unit.is_leader() and deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            return self.unit.is_leader() and (
+                deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+                or deployment_desc.typ == DeploymentType.OTHER
+                and "data" in deployment_desc.config.roles
```

Comment:
I believe this check is too broad, as it also impacts subsequent data units / clusters way past the sec index initialisation